src/dotnet/commands/dotnet-compile/CompilerUtil.cs

```diff
@@ -47,13 +47,26 @@ public NonCultureResgenIO(string inputFile, string outputFile, string metadataNa
         }
 
         // used in incremental compilation
-        public static List<NonCultureResgenIO> GetNonCultureResources(Project project, string intermediateOutputPath)
+        public static List<NonCultureResgenIO> GetNonCultureResources(Project project, string intermediateOutputPath, CommonCompilerOptions compilationOptions)
         {
-            return 
-                (from resourceFile in project.Files.ResourceFiles
-                    let inputFile = resourceFile.Key
+            if (compilationOptions.EmbedInclude == null)
+            {
+                return
+                    (from resourceFile in project.Files.ResourceFiles
+                        let inputFile = resourceFile.Key
+                        where string.IsNullOrEmpty(ResourceUtility.GetResourceCultureName(inputFile))
+                        let metadataName = GetResourceFileMetadataName(project, resourceFile.Key, resourceFile.Value)
+                        let outputFile = ResourceUtility.IsResxFile(inputFile) ? Path.Combine(intermediateOutputPath, metadataName) : null
+                        select new NonCultureResgenIO(inputFile, outputFile, metadataName)
+                        ).ToList();
+            }
+
+            return
```

Comment:
there is no reason to reuse this method. create a overload and run different logic based on if the `compilationOptions` is given.